etcdserver/server.go

```diff
@@ -1988,6 +1992,67 @@ func (s *EtcdServer) sync(timeout time.Duration) {
 	})
 }
 
+// publishV3 registers server information into the cluster using v3 request. The
+// information is the JSON representation of this server's member struct, updated
+// with the static clientURLs of the server.
+// The function keeps attempting to register until it succeeds,
+// or its server is stopped.
+func (s *EtcdServer) publishV3(timeout time.Duration) {
+	req := &membershippb.ClusterMemberAttrSetRequest{
+		Member_ID: uint64(s.id),
+		MemberAttributes: &membershippb.Attributes{
+			Name:       s.attributes.Name,
+			ClientUrls: s.attributes.ClientURLs,
+		},
+	}
+
+	for {
+		select {
+		case <-s.stopping:
+			if lg := s.getLogger(); lg != nil {
```

Comment:
Can we update `getLogger` to return `zap.NewNop()`, so we don't need all these manual `nil` checks?

Reply:
Done. Will fix other `getLogger` in a separate pr.